packages/clerk-js/src/core/clerk.ts

```diff
@@ -831,10 +834,40 @@ export class Clerk implements ClerkInterface {
     if (beforeEmit) {
       beforeUnloadTracker?.startTracking();
       this.#setTransitiveState();
+      deprecated('beforeEmit', 'Use the `redirectUrl` property instead.');
       await beforeEmit(newSession);
       beforeUnloadTracker?.stopTracking();
     }
 
+    if (redirectUrl) {
+      beforeUnloadTracker?.startTracking();
+      this.#setTransitiveState();
+
+      if (
+        this.client.cookieExpiresAt &&
+        this.client.cookieExpiresAt.getTime() - Date.now() <= 8 * 24 * 60 * 60 * 1000 // 8 days
+      ) {
```

Comment:
❓ Would it be possible to replace this check with data on `Clerk.client` so this redirect logic is controlled by FAPI instead? 

I think that we could populate a `client` field if `clerk-js` had to refresh the `client` cookie without adding logic to clerk-js itself, except for the actual redirect, of course.

Reply:
Good question! I've introduced a method on the client (`isEligibleForTouch`) which will perform this check and simplify the statement here based on the suggestion below.

Let me know if that's in line with what you had in mind 😊